pkg/nodebootstrap/managed_al2_test.go

```diff
@@ -0,0 +1,84 @@
+package nodebootstrap_test
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+
+	"github.com/weaveworks/eksctl/pkg/nodebootstrap"
+)
+
+type managedEntry struct {
+	ng *api.ManagedNodeGroup
+
+	expectedUserData string
+}
+
+var _ = DescribeTable("Managed AL2", func(e managedEntry) {
+	api.SetManagedNodeGroupDefaults(e.ng, &api.ClusterMeta{Name: "cluster"})
+	bootstrapper := nodebootstrap.NewManagedAL2Bootstrapper(e.ng)
+	bootstrapper.UserDataMimeBoundary = "//"
+	userData, err := bootstrapper.UserData()
+	Expect(err).ToNot(HaveOccurred())
+	Expect(userData).To(Equal(e.expectedUserData))
+},
+	Entry("SSM enabled", managedEntry{
+		ng: &api.ManagedNodeGroup{
+			NodeGroupBase: &api.NodeGroupBase{
+				Name: "ssm-enabled",
+				SSH: &api.NodeGroupSSH{
+					Allow:         api.Enabled(),
+					PublicKeyName: aws.String("test-keypair"),
+					EnableSSM:     api.Enabled(),
+				},
+			},
+		},
+
+		expectedUserData: `TUlNRS1WZXJzaW9uOiAxLjANCkNvbnRlbnQtVHlwZTogbXVsdGlwYXJ0L21peGVkOyBib3VuZGFyeT0vLw0KDQotLS8vDQpDb250ZW50LVR5cGU6IHRleHQveC1zaGVsbHNjcmlwdA0KQ29udGVudC1UeXBlOiBjaGFyc2V0PSJ1cy1hc2NpaSINCg0KIyEvYmluL2Jhc2gKCnNldCAtbyBlcnJleGl0CnNldCAtbyBwaXBlZmFpbApzZXQgLW8gbm91bnNldAoKeXVtIGluc3RhbGwgLXkgYW1hem9uLXNzbS1hZ2VudApzeXN0ZW1jdGwgZW5hYmxlIGFtYXpvbi1zc20tYWdlbnQKc3lzdGVtY3RsIHN0YXJ0IGFtYXpvbi1zc20tYWdlbnQKDQotLS8vLS0NCg==`,
```

Comment:
Can we decode the userdata and check against human readable fields?

Reply:
Yeah, that's how I wrote it initially but it involved replacing CRLF sequences. I agree that it's much more readable than encoded values. I'll see how we can best express it.